OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -88,6 +89,12 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+
+		public override TargetLineNode TargetLineNode(Actor self)
+		{
+			var color = NextActivity == null ? Color.Green : NextActivity.TargetLineNode(self).Color;
```

Comment:
This is going to hurt performance and maintainability.  Each target line needs to return the properties of *itself*, without depending on what the next segment is doing.  When you draw the lines you can then look at the properties of the last node to get the color to draw the whole chain.